var/spack/repos/builtin/packages/qgis/package.py

```diff
@@ -0,0 +1,214 @@
+    depends_on('qca@2.2.1')
+    depends_on('py-pyqt4 +qsci', when='@2')
+    depends_on('py-pyqt5@5.3: +qsci', when='@3')
+    depends_on('qscintilla')
+    depends_on('qjson')
+    depends_on('py-requests', type=('build', 'run'))  # TODO: is build dependency necessary?
+    depends_on('py-psycopg2', type=('build', 'run'))  # TODO: is build dependency necessary?
+    depends_on('qtkeychain@0.5:', when='@3:')
+    depends_on('libzip')
+    depends_on('exiv2')
+    depends_on('python@3.0.0:', type=('build', 'run'), when='@3')
+    depends_on('python@2.7:2.8', type=('build', 'run'), when='@2')
+
+    # Runtime python dependencies, not mentioned in install instructions
+    depends_on('py-pyyaml', type='run')
+    depends_on('py-owslib', type='run')
+    depends_on('py-jinja2', type='run')
+    depends_on('py-pygments', type='run')
+
+    # optionals
+    depends_on('postgresql@8:', when='+postgresql')  # for PostGIS support
+    depends_on('gsl', when='+georeferencer')  # for georeferencer
+    depends_on('grass@7.0.0', type=('build', 'link', 'run'), when='+grass7')  # for georeferencer
+
+    # the below dependencies are shown in cmake config
+    depends_on('hdf5')
+    depends_on('netcdf-c')
+
+    # build
+    depends_on('cmake@3.0.0:', type='build')
+    depends_on('flex@2.5.6:', type='build')
+    depends_on('bison@2.4:', type='build')
+    depends_on('pkg-config', type='build')
+
+    # Conflicts for newer versions
+    conflicts('proj@:4.9.2', when='@3.8.2:')
+
+    # v3.8.1, Qt >= 5.9.0 is required
+    conflicts('qt@:5.8.99', when='@3.8.1:')
+    conflicts('qt@5.13.0:', when='@3.8.1')
+
+    # conflicts for qgis@2, qt@4, python@2
+    conflicts('qtkeychain@0.6.0:', when='^qt@4')
+    conflicts('qt@5:', when='@2')
+
+    def cmake_args(self):
+        spec = self.spec
+        args = []
+        # qtwebkit module was removed from qt as of version 5.6
+        # needs to be compiled as a separate package
+        args.extend(['-DUSE_OPENCL=OFF',
+            # cmake couldn't determine the following paths
+            '-DEXPAT_LIBRARY={0}'.format(self.spec['expat'].libs),
+            '-DPOSTGRESQL_PREFIX={0}'.format(self.spec['postgresql'].prefix),
+            '-DQSCINTILLA_INCLUDE_DIR=' +
+                str(self.spec['qscintilla'].prefix) + '/include',
+            '-DQSCINTILLA_LIBRARY=' + str(self.spec['qscintilla'].prefix) +
+                '/lib/libqscintilla2_qt5.so',
+            '-DLIBZIP_INCLUDE_DIR=' +
+                str(self.spec['libzip'].prefix) + '/include',
+            '-DLIBZIP_CONF_INCLUDE_DIR=' +
+                str(self.spec['libzip'].prefix) + '/lib/libzip/include',
+            '-DGDAL_CONFIG_PREFER_PATH=' + str(self.spec['gdal'].prefix.bin),
+            '-DGEOS_CONFIG_PREFER_PATH=' + str(self.spec['geos'].prefix.bin),
+            '-DGSL_CONFIG_PREFER_PATH=' + str(self.spec['gsl'].prefix.bin),
+            '-DPOSTGRES_CONFIG_PREFER_PATH=' +
+                str(self.spec['postgresql'].prefix.bin)
```

Comment:
All of these `Prefix` objects already subclass `str`, so you shouldn't need to cast them to `str`

Reply:
Good to know.